src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1588,13 +1588,27 @@ public override BoundNode VisitCall(BoundCall node)
                 ImmutableArray<RefKind> refKindsOpt = node.ArgumentRefKindsOpt;
                 ImmutableArray<BoundExpression> arguments = RemoveArgumentConversions(node.Arguments, refKindsOpt);
                 ImmutableArray<int> argsToParamsOpt = node.ArgsToParamsOpt;
-                ImmutableArray<Result> results = VisitArgumentsEvaluate(arguments, refKindsOpt, method.Parameters, argsToParamsOpt, node.Expanded);
+                // We do a first pass to work through the arguments without making any assumptions
+                ImmutableArray<Result> results = VisitArgumentsEvaluate(arguments, refKindsOpt);
 
                 if (method.IsGenericMethod && HasImplicitTypeArguments(node))
                 {
                     method = InferMethod(node, method, results.SelectAsArray(r => r.Type));
                 }
-                VisitArgumentsWarn(arguments, refKindsOpt, method.Parameters, argsToParamsOpt, node.Expanded, results);
+
+                VisitArgumentConversions(arguments, refKindsOpt, method.Parameters, argsToParamsOpt, node.Expanded, results);
```

Comment:
Can `VisitArgumentConversion` be moved after `VisitArgumentsEvaluateHonoringAnnotations`? Then, can `VisitArgumentsEvaluateHonoringAnnotations` be moved inside `VisitArgumentEvaluate`?